system/ee/ExpressionEngine/Addons/spam/upd.spam.php

```diff
@@ -476,6 +482,39 @@ private function updateForumSpam_2_00_00()
             $trap->save();
         }
     }
+
+    /** Add comments with SPAM status to the trap */
+    private function trapSpamComments()
+    {
+        // get the comments so we can save a serialized entity model to the spam trap
+        $spam_comments = ee('Model')->get('Comment')
+            ->filter('status', 's');
+        if ($spam_comments->count() == 0) {
+            return;
+        }
+
+        $trapped_comments = ee('Model')->get('spam:SpamTrap')
+            ->filter('content_type', 'comment');
+
+        $trappedCommentIds = array();
+        if ($trapped_comments->count() > 0) {
+            foreach ($trapped_comments->all() as $trapped) {
+                $trappedCommentIds[] = $trapped->comment_id;
+            }
+        }
+
+        if (!empty($trappedCommentIds)) {
+            $spam_comments->filter('comment_id', 'NOT IN', $trappedCommentIds);
+        }
+
+        if ($spam_comments->count() == 0) {
+            return;
+        }
+
+        foreach ($spam_comments->all() as $comment) {
```

Comment:
@intoeetive this could potentially be a large dataset so we might want to consider iterating through smaller chunks of retrieved models